CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+      puts 'Not again!'
+    when song.duration > 120
+      puts 'Too long!'
+    when Time.now.hour > 21
+      puts "It's too late"
+    else
+      song.play
+  end
+
+  # good
+  case
+  when song.name == 'Misty'
+    puts 'Not again!'
+  when song.duration > 120
+    puts 'Too long!'
+  when Time.now.hour > 21
+    puts "It's too late"
+  else
+    song.play
+  end
+  ```
+
+* When assigning the result of a conditional expression to a variable, preserve the usual alignment of its branches.
+
+  ```Ruby
+  # bad - pretty convoluted
+  kind = case year
+  when 1850..1889 then 'Blues'
+  when 1890..1909 then 'Ragtime'
+  when 1910..1929 then 'New Orleans Jazz'
+  when 1930..1939 then 'Swing'
+  when 1940..1950 then 'Bebop'
+  else 'Jazz'
+  end
+
+  result = if some_cond
+    calc_something
+  else
+    calc_something_else
+  end
+
+  # good - it's apparent what's going on
+  kind = case year
+         when 1850..1889 then 'Blues'
+         when 1890..1909 then 'Ragtime'
+         when 1910..1929 then 'New Orleans Jazz'
+         when 1930..1939 then 'Swing'
+         when 1940..1950 then 'Bebop'
+         else 'Jazz'
+         end
+
+  result = if some_cond
+             calc_something
+           else
+             calc_something_else
+           end
+
+  # good (and a bit more width efficient)
+  kind =
+    case year
+    when 1850..1889 then 'Blues'
+    when 1890..1909 then 'Ragtime'
+    when 1910..1929 then 'New Orleans Jazz'
+    when 1930..1939 then 'Swing'
+    when 1940..1950 then 'Bebop'
+    else 'Jazz'
+    end
+
+  result =
+    if some_cond
+      calc_something
+    else
+      calc_something_else
+    end
+  ```
+
+* Use empty lines between method definitions and also to break up a method into logical paragraphs internally.
+
+  ```Ruby
+  def some_method
+    data = initialize(options)
+
+    data.manipulate!
+
+    data.result
+  end
+
+  def some_method
+    result
+  end
+  ```
+
+* Avoid comma after the last parameter in a method call, especially when the parameters are not on separate lines.
+
+  ```Ruby
+  # bad - easier to move/add/remove parameters, but still not preferred
+  some_method(
+               size,
+               count,
+               color,
+             )
+
+  # bad
+  some_method(size, count, color, )
+
+  # good
+  some_method(size, count, color)
+  ```
+
+* Avoid line continuation `\` where not required. In practice, avoid using line continuations for anything but string concatenation.
+
+  ```Ruby
+  # bad
+  result = 1 - \
+           2
+
+  # good (but still ugly as hell)
+  result = 1 \
+           - 2
+
+  long_string = 'First part of the long string' \
+                ' and second part of the long string'
+  ```
+
+* Adopt a consistent multi-line method chaining style. There are two popular styles in the Ruby community, both of which are considered good - leading `.` (Option A) and trailing `.` (Option B).
+
+  * **(Option A)** When continuing a chained method invocation on
```

Comment:
We should only have 1 option documented as our style, here. In this case, my preference is always to have an operator lead a continued-line because we read left-to-right and the start of the continued-line begins with an operator so we can assume it starts from the previous line. This is `Option A` in this example.